Clarify whether CRDs are cluster scoped in multitenancy page (#48156)
kubernetes/website: content/en/docs/concepts/security/multi-tenancy.md

Namespace isolation is well-supported by Kubernetes, has a negligible resource cost, and provides
mechanisms to allow tenants to interact appropriately, such as by allowing service-to-service

> communication. However, it can be difficult to configure, and it doesn't apply to resources that are cluster-scoped and not namespaced, such as Custom Resource Definitions (CRDs) themselves (though the custom resources they define can be namespaced), Storage Classes, and Webhooks.

Reviewer: When we edit existing text, it's good to follow our style guide (no need to fix the rest of the page, although those fixes are also welcome).
Try this:
```suggestion
communication. However, it can be difficult to configure, and it doesn't apply to Kubernetes resources
that aren't namespaced, such as APIServices, StorageClasses, or ClusterTrustBundles.
```

(there's no Webhook API, StorageClass doesn't have a space, and CRD is a poor choice of example as it can mislead)

Author: Thank you for the feedback, @sftim I've made the suggested changes to align with the Kubernetes style guide. The examples now use APIServices, StorageClasses, and ClusterTrustBundles instead of CRDs and Webhooks to avoid confusion.

Let me know if further adjustments are needed!

Namespace isolation is well-supported by Kubernetes, has a negligible resource cost, and provides
mechanisms to allow tenants to interact appropriately, such as by allowing service-to-service
communication. However, it can be difficult to configure, and it doesn't apply to Kubernetes resources

> that aren't namespaced, such as APIServices, StorageClasses, or ClusterTrustBundles.

Reviewer 2: Since the confusion is common to some users, we should not shy away from mentioning CustomResourceDefinitions. We may instead want to emphasize that CustomResourceDefinitions are cluster scoped, but custom resources created from CRDs do not necessarily have to be cluster scoped.
Otherwise, what's the point of this PR then?

Author: Thank you for the feedback, @tengqm and @sftim! I've updated the statement to maintain mention of CustomResourceDefinitions (CRDs) for clarity, specifying that CRDs are cluster-scoped, but the custom resources they define can be namespaced. Here's the updated text:

"However, it can be difficult to configure, and it doesn't apply to Kubernetes resources that aren't namespaced, such as APIServices, StorageClasses, ClusterTrustBundles, or CustomResourceDefinitions (CRDs) themselves, though the custom resources they define can be namespaced."

Let me know if this looks good, and I'll proceed with the updates.

> that aren't namespaced, such as APIServices, StorageClasses, ClusterTrustBundles, or CustomResourceDefinitions (CRDs) themselves, though the custom resources they define can be namespaced.

Reviewer 2: ```suggestion
that aren't namespaced, such as APIServices, StorageClasses, ClusterTrustBundles.
Note that CustomResourceDefinitions (CRDs) are cluster scoped
though the custom resources they define can be namespaced.
```

Author: Fix the issue
